content/blogposts/2022/release-4.2.md

```diff
@@ -0,0 +1,87 @@
+
+This feature is in Beta, and feedback is very welcome. Tweet at us, or drop a comment in this [issue](https://github.com/sourcegraph/sourcegraph/issues/44597)!
+
+<br />
+<Badge link="/code-search" text="Code Search" color="cerise" size="small" />
+
+#### A Smart Search toggle to assist with search queries
+
+The new lightning bolt toggle <span style={{display: "inline-flex", verticalAlign: "middle", margin: "2px"}}><img style={{width: "18px", height: "18px"}} src="https://storage.googleapis.com/sourcegraph-assets/about.sourcegraph.com/blog/2022/smart-search-bar-lightning.png"/></span> in the search bar activates `Smart Search`. Smart Search is a query assistant that activates when a search ordinarily returns no results. 
+
+The basic idea is to help find search results that are likely to be more useful than showing "no results" by trying slight variations of the original query. Smart Search works by trying alternative queries based on a handful of rules (we know how easy it is to get tripped up by query syntax). When a query alternative finds results, those results are shown immediately.
+
+Take a query like `go buf byte parser`, for example. Normally, Sourcegraph will search for the string "go buf byte parser" with those tokens in that order. If there are **_no_** results, Smart Search attempts variations of the query. One rule applies a `lang:` filter to known languages. For example, `go` may refer to the `Go` language, so we convert this token to a `lang:Go` filter. Additionally, another rule relaxes the ordering on remaining tokens so that we search for `buf AND byte AND parser` anywhere in the file. Here's an example of what Smart Search looks like in action:
+
+<img src="https://storage.googleapis.com/sourcegraph-assets/about.sourcegraph.com/blog/2022/smart-search-example.png"/>
+
+Note that if the original query found results (which depends on the code it runs on), Smart Search has no effect. Smart Search does not otherwise intervene (or interfere with) search queries if those queries return results, and Sourcegraph behaves as usual. 
+
+It is sometimes useful to check for the _absence_ of results (we _want_ to see zero matches). In these cases Smart Search can disabled temporarily by toggling the button in the search bar. To disable Smart Search permanently by default, set `"search.defaultMode": "precise"` in settings.
+
+It is not possible to customize Smart Search rules at this time. So far a small number of rules are enabled based on feedback and utility. They affect the following query properties:
+
+- Separate patterns with `AND` (pattern order doesn't matter)
+- Patterns as filters (e.g., apply `lang:` or `type:symbol`  filters based on keywords)
+- Quotes in queries (run a literal search for quoted patterns)
+- Patterns as Regular Expressions (check patterns for likely regular expression syntax)
```

Comment:
@iskyOS feel free to cut/edit/arrange as you like!

Reply:
yes, good point. I can factor it out. Since it looks like we are pushing put the release cut (and so this post too), I should have time to do that today. Sound good?

Reply:
Yes, sounds good to me!